Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean

```diff
@@ -927,8 +928,71 @@ def localizationMapOfSpecializes {x y : PrimeSpectrum R} (h : x ⤳ y) :
         ⟨a, show a ∈ x.asIdeal.primeCompl from h ha⟩ : _))
 #align prime_spectrum.localization_map_of_specializes PrimeSpectrum.localizationMapOfSpecializes
 
+variable (R) in
+/--
+Zero loci of prime ideals are closed irreducible sets in the Zariski topology and any closed
+irreducible set is a zero locus of some prime ideal.
+-/
+protected def pointsEquivIrreducibleCloseds :
+    PrimeSpectrum R ≃o {s : Set (PrimeSpectrum R) | IsIrreducible s ∧ IsClosed s}ᵒᵈ where
+  __ := irreducibleSetEquivPoints.toEquiv.symm.trans OrderDual.toDual
+  map_rel_iff' {p q} :=
+    (RelIso.symm irreducibleSetEquivPoints).map_rel_iff.trans (le_iff_specializes p q).symm
+
 end PrimeSpectrum
 
+open PrimeSpectrum in
+/--
+[Stacks: Lemma 00ES (3)](https://stacks.math.columbia.edu/tag/00ES)
+Zero loci of minimal prime ideals of `R` are irreducible components in `Spec R` and any
+irreducible component is a zero locus of some minimal prime ideal.
+-/
+protected def minimalPrimes.equivIrreducibleComponents :
+    minimalPrimes R ≃o (irreducibleComponents <| PrimeSpectrum R)ᵒᵈ :=
+  let e := PrimeSpectrum.pointsEquivIrreducibleCloseds R
+  { toFun := fun p ↦
+      let s := e ⟨p.1, p.2.1.1⟩
+      OrderDual.toDual ⟨s.1, ⟨s.2.1, fun t (ht : IsIrreducible t) (le : closure _ ⊆ t) ↦
+        show t ⊆ closure _ from PrimeSpectrum.closure_singleton ⟨p.1, _⟩ ▸ by
+          exact subset_zeroLocus_iff_le_vanishingIdeal t p.1 |>.mpr <| p.2.2
+            ⟨isIrreducible_iff_vanishingIdeal_isPrime.mp ht, bot_le⟩ fun x hx ↦
+              (mem_vanishingIdeal _ _ |>.mp hx) ⟨p.1, p.2.1.1⟩ <| le <|
+                closure_singleton ⟨p.1, _⟩ ▸ (mem_zeroLocus _ _ |>.mpr <| le_refl _)⟩⟩
+    invFun := fun s ↦
+      let p := e.symm <| OrderDual.toDual ⟨s.1, s.2.1, isClosed_of_mem_irreducibleComponents _ s.2⟩
+      ⟨p.1, ⟨p.2, bot_le⟩, fun q ⟨hq, _⟩ (le : q ≤ s.2.1.genericPoint.asIdeal) ↦
+        show s.2.1.genericPoint.asIdeal ≤ q by
+        have eq1 : closure _ = _ := s.2.1.genericPoint_spec
+        rw [isClosed_of_mem_irreducibleComponents _ s.2 |>.closure_eq,
+          PrimeSpectrum.closure_singleton] at eq1
+        rw [← hq.radical, ← zeroLocus_subset_zeroLocus_iff, eq1]
+        exact s.2.2 (isIrreducible_zeroLocus_iff_of_radical _ hq.isRadical |>.mpr hq) <|
+          subset_zeroLocus_iff_le_vanishingIdeal _ _ |>.mpr <| le.trans <| by
+          rw [← subset_zeroLocus_iff_le_vanishingIdeal, eq1]⟩
+    left_inv := fun p ↦ Subtype.ext <| show (e.symm <| e _).1 = p.1 by
+      rw [e.symm_apply_apply]
+    right_inv := fun s ↦ Subtype.ext <| show (e (e.symm _)).1 = s.1 by
+      rw [e.apply_symm_apply]; rfl
+    map_rel_iff' := e.map_rel_iff }
+
+open PrimeSpectrum in
+lemma minimalPrimes.eq_irreducibleComponents :
+    minimalPrimes R =
+    PrimeSpectrum.vanishingIdeal '' (irreducibleComponents <| PrimeSpectrum R) := by
```

Comment:
It might be more useful to show:
+ The vanishing ideal of a closed set is a minimal prime iff the closed set is an irreducible component
+ The zero locus of an ideal is an irreducible component iff the ideal is a minimal prime
and then you can derive this easily.

Reply:
>     * The zero locus of an ideal is an irreducible component iff the ideal is a minimal prime
I could only show that the zero locus of an ideal `I` is an irreducible component iff there exists a minimal ideal `J` with the same zero locus as `I`

Reply:
Oh yes, radicality is necessary. Thanks!

Reply:
The assumption of radicalness and closedness make the two lemmas hard to apply in this equality lemma.

Reply:
Following the convention of putting the more complex term on the LHS, maybe we should flip this and call it `PrimeSpectrum.vanishingIdeal_irreducibleComponents`?

Reply:
done